### PR TITLE
Enable complex in gen_* scripts

### PR DESCRIPTION
**Pull request by rasolca**
Added the possibility to choose the type in `miniapp.py`

Updated the `gen_dlaf_*.py` scripts accordingly
Added no-random tridiagonal eigensolver runs directly to gen_dlaf_strong_*.py`
Added slate eigensolver as well

### CODE REVIEW

File: scripts/miniapps.py [hunk 1]
```diff
@@ -168,28 +183,33 @@ def chol(
     suffix="na",
     extra_flags="",
     env="",
+    typ="d",
```
Comment (msimberg) [inline scripts/miniapps.py:186]: Is `type` reserved for something as a name?

File: scripts/gen_dlaf_weak-gpu.py [hunk 2]
```diff
@@ -24,22 +24,22 @@
 
 run_dir = "~/ws/runs/weak"
 
-
+# Note: job time is computed as time0 + sqrt(nodes) * time
+time0 = 120  # minutes
+time = 0  # minutes
```
Comment (msimberg) [inline scripts/gen_dlaf_weak-gpu.py:29]: Not asking for a change here, but I never noticed before: Why is this 0?

**Review by msimberg — APPROVED** [review]

File: scripts/miniapps.py (hunk 1 above)
Comment (rasolca) [reply scripts/miniapps.py:186]: https://www.w3schools.com/python/ref_func_type.asp

Comment (msimberg) [reply scripts/miniapps.py:186]: :sob: 

`datatype`? Not insisting on this at all (as you can see from my approval).

File: scripts/gen_dlaf_strong-gpu.py [hunk 3]
```diff
@@ -45,99 +47,104 @@
...
 
     run.submit(run_dir, debug=debug)
 
+    run = mp.StrongScaling(system, "DLAF_test_strong", "job_dlaf-norandom", nodes_arr, time)
+    for m_sz in m_szs:
+        trid_kwargs = full_kwargs.copy()
+        trid_kwargs["suffix"] = "fromfile"
+        trid_kwargs["extra_flags"] = (
+            trid_kwargs.get("extra_flags", "")
+            + f" --input-file=/scratch/e1000/rasolca/trid-ref-{m_sz}.h5"
```
Comment (albestro) [inline scripts/gen_dlaf_strong-gpu.py:133]: it seems a bit too hard-coded 🤣

File: scripts/miniapps.py (hunk 1 above)
Comment (albestro) [reply scripts/miniapps.py]: https://github.com/eth-cscs/DLA-Future/pull/1010/commits/4425298ac53c5e5684e15dfb2bc6432de1db4a10

**Pull request merged**